python_modules/dagster/dagster/_core/scheduler/instigation.py

```diff
@@ -241,6 +291,18 @@ def with_origin_run(self, origin_run_id: str) -> "InstigatorTick":
     def with_log_key(self, log_key: Sequence[str]) -> "InstigatorTick":
         return self._replace(tick_data=self.tick_data.with_log_key(log_key))
 
+    def with_dynamic_partitions_request_result(
+        self,
+        dynamic_partitions_request_result: Union[
+            AddDynamicPartitionsRequestResult, DeleteDynamicPartitionsRequestResult
```

Comment:
consider combining the two into just a `DynamicPartitionsRequestResult` to reduce boiler plate / unnecessary type checking?  It looks like it's just for display purposes...  We could check that there are either additions or deletions, not both.

Reply:
good call